qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -0,0 +1,115 @@
+        The mode |k> is then mapped to the occupation number vector
+        |0_{nk_^max}, 0_{nk_^max - 1}, ..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1}, ..., 0_{0_k}>
+    """
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp | PauliSumOp:
+        # If register_length was passed, override the one present in the BosonicOp
+        if register_length is None:
+            register_length = second_q_op.num_modes
+
+        qubit_register_length = register_length * (self.truncation + 1)
+        # Create a Pauli operator, which we will fill in this method
+        # TODO: Check if we need to normal/index_order
+        pauli_op: list[SparsePauliOp] = []
+        # Then we loop over all the terms of the bosonic operator
+        for terms, coeff in second_q_op.terms():
+            # Then loop over each term (terms -> List[Tuple[string, int]])
+            bos_op_to_pauli_op = SparsePauliOp(["I" * qubit_register_length], coeffs=[1.])
+            for op, idx in terms:
+                if op not in ('+', '-'):
+                    break
+                pauli_expansion: list[SparsePauliOp] = []
+                # Now we are dealing with a single bosonic operator. We have to perform the linear mapper
+                for n_k in range(self.truncation):
+                    prefactor = np.sqrt(n_k + 1) / 4
+                    # Define the actual index in the qubit register. It is given n_k plus the shift due
+                    # to the mode onto which the operator is acting
+                    register_index = n_k + idx * (self.truncation + 1)
+                    # Now build the Pauli operators XX, XY, YX, YY, which arise from S_i^+ S_j^-
+                    x_x, x_y, y_x, y_y = self.get_ij_pauli_matrix(register_index, qubit_register_length)
+
+                    tmp_op = SparsePauliOp(x_x) + SparsePauliOp(y_y)
+                    if op == "+":
+                        tmp_op += - 1j*SparsePauliOp(x_y) + 1j*SparsePauliOp(y_x)
+                    else:
+                        tmp_op += + 1j*SparsePauliOp(x_y) - 1j*SparsePauliOp(y_x)
+                    pauli_expansion.append(prefactor * tmp_op)
+                # Add the Pauli expansion for a single n_k to map of the bosonic operator
+                bos_op_to_pauli_op = reduce(operator.add, pauli_expansion).compose(bos_op_to_pauli_op)
+            # Add the map of the single boson op (e.g. +_0) to the map of the full bosonic operator
+            pauli_op.append(coeff * reduce(operator.add, bos_op_to_pauli_op.simplify()))
+
+        # return the lookup table for the transformed XYZI operators
+        bos_op_encoding = reduce(operator.add, pauli_op)
+        return bos_op_encoding
+
+    def get_ij_pauli_matrix(self, register_index: int, register_length: int):
+        """This method builds the Qiskit Pauli operators of the operators XX, YY, XY and YX"""
```

Comment:
Template suggestion to address comments above (needs to import `from functools import lru_cache` at the top)

```suggestion
    @classmethod
    @lru_cache(maxsize=32)
    def _get_ij_pauli_matrix(register_index: int, register_length: int):
        """This method builds the Qiskit Pauli operators of the operators XX, YY, XY and YX
               Args:
                   ...
               Returns:
                   ...
        """
```